dotCMS/src/main/java/com/dotcms/concurrent/DotConcurrentFactory.java

```diff
@@ -578,6 +656,91 @@ public void run() {
         }
     } // DelayQueueConsumer.
 
+    /// DotSingleSubmitterImpl
+    private final class DotSingleSubmitterImpl implements DotSubmitter {
+
+        private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+
+        @Override
+        public Future<?> submit(final Runnable command) {
+            return this.executorService.submit(command);
+        }
+
+        @Override
+        public void delay(final Runnable task, final long delay, final TimeUnit unit) {
+
+            throw new UnsupportedOperationException("Delay not supported on single submitter");
+        }
+
+        @Override
+        public Future<?> submit(final Runnable command, final long delay, final TimeUnit unit) {
+
+            throw new UnsupportedOperationException("Submit Delay not supported on single submitter");
+        }
+
+        @Override
+        public <T> Future<T> submit(final Callable<T> callable) {
+
+            return this.executorService.submit(callable);
+        }
+
+        @Override
+        public <T> Future<T> submit(Callable<T> callable, long delay, TimeUnit unit) {
+
+            throw new UnsupportedOperationException("Submit Delay not supported on single submitter");
```

Comment:
Is it possible to add the name of the submitter in the error msg? For troubleshooting/support purposes.